### PR TITLE
Added canceled_on field to unified_jobs model

### PR DESCRIPTION
**Pull request by thedoubl3j**
- When a job is canceled, the canceled_on field will populate with date/time
related #3988
##### SUMMARY

Please see #5406 for previous comments/discussion about the decisions made. 

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 9.1.0

```

##### ADDITIONAL INFORMATION

Before, this would look like

    "scm_type": "git",
    "scm_url": "https://github.com/ansible/test-playbooks.git",
    "scm_branch": "",
    "scm_clean": false,
    "scm_delete_on_update": false,
    "credential": null,
    "timeout": 0,
    "unified_job_template": 3838,
    "launch_type": "manual",
    "status": "canceled",
    "failed": true,
    "started": null,
    "finished": "2019-06-03T12:39:05.272329Z",
    "elapsed": 0.0,
    "job_args": "",
    "job_cwd": "",
    "job_env": {},
    "job_explanation": "",
    "execution_node": "",
    "result_traceback": "",
    "event_processing_finished": true,
    "project": 3838,
    "job_type": "check",
With this addition, a new field is added called canceled_on that displays the timestamp that the job was canceled based on the cancel_flag. 

So it now will display something like 

```
"created": "2020-01-07T19:37:08.997962Z",
            "modified": "2020-01-07T19:37:09.325185Z",
            "name": "test",
            "description": "",
            "unified_job_template": 9,
            "launch_type": "relaunch",
            "status": "failed",
            "failed": true,
            "started": "2020-01-07T19:37:09.491372Z",
            "finished": "2020-01-07T19:48:25.446239Z",
            "canceled_on": "2020-01-07T19:55:01.833860Z",
            "elapsed": 675.955,
            "job_explanation": "Task was marked as running in Tower but was not present in the job queue, so it has been marked as failed.",
            "execution_node": "awx",
```

### CODE REVIEW

**Comment by thedoubl3j** [comment]
a big thanks _again_ to @rooftopcellist @beeankha and @rebeccahhh for the help provided.

File: awx/main/models/unified_jobs.py [hunk 1]
```diff
@@ -810,11 +817,18 @@ def save(self, *args, **kwargs):
 
         # Sanity check: Has the job just completed? If so, mark down its
         # completion time, and record its output to the database.
+        #import sdb
```
Comment (ryanpetrello) [inline awx/main/models/unified_jobs.py]: Let's remove this.

**Review by ryanpetrello — COMMENTED** [review]
This seems pretty reasonable to me.

@mabashian @jakemcdermott can you think of anything we might want to do with this in the UI?

**Comment by thedoubl3j** [comment]
I actually have a problem with it @ryanpetrello because this...is wrong

File: awx/main/models/unified_jobs.py [hunk 2]
```diff
@@ -630,6 +630,13 @@ class Meta:
         help_text=_("The date and time the job finished execution."),
         db_index=True,
     )
+    canceled_on = models.DateTimeField(
+        null=True,
+        default=None,
+        editable=False,
+        help_text=_("The date and time the job was canceled."),
```
Comment (AlanCoding) [inline awx/main/models/unified_jobs.py]: might be better to say "the date and time of the request ...", but different people have different ideas about how general vs. accurate the OPTIONS help should be.

**Review by AlanCoding — APPROVED** [review]
I love how simple this is. It needs to be in the serializer, but if it's easier to remove it from the summary_fields data to get tests passing, I would not object to that.

**Review by ryanpetrello — APPROVED** [review]
It looks like Alan has some minor feedback, but this looks great 🎉

Thanks @thedoubl3j !

**Comment by AlanCoding** [comment]
My comments were only suggestions to get the test passing, that's the only blocker

File: awx/main/migrations/0101_v370_unifiedjob_canceled.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('main', '0100_v370_projectupdate_job_tags'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='unifiedjob',
+            name='canceled_on',
+            field=models.DateTimeField(db_index=True, default=None, editable=False, help_text='The date and time the job was canceled.', null=True),
```
Comment (AlanCoding) [inline awx/main/migrations/0101_v370_unifiedjob_canceled.py]: this won't match the model field's help text anymore

Comment (thedoubl3j) [reply awx/main/migrations/0101_v370_unifiedjob_canceled.py]: oof, sorry, should have made that change while I was making the other. one sec and I will add it here as well.

**Comment by thedoubl3j** [comment]
@AlanCoding so what do you think the next steps are? anything else we should add/remove/change?

**Comment by AlanCoding** [comment]
I think you need to rebase to get tests passing

**Pull request merged**